src/util/strb.c

```diff
@@ -34,24 +34,35 @@ int strb_grow(strb *sb, size_t n) {
   return 0;
 }
 
-void strb_appendf(strb *sb, const char *f, ...) {
-  va_list ap;
-  int s;
+void strb_appendv(strb *sb, const char *f, va_list ap) {
+  va_list apSave;
+  int     s;
 
-  va_start(ap, f);
 #ifdef _MSC_VER
-  s = _vscprintf(f, ap);
+  /**
+   * va_copy() is a C99 novelty that a particular company should have started
+   * supporting a long time ago, to their undying shame.
+   */
+  
+  apSave = ap;
+  s = _vscprintf(f, apSave);
 #else
-  s = vsnprintf(NULL, 0, f, ap);
+  va_copy(apSave, ap);
+  s = vsnprintf(NULL, 0, f, apSave);
 #endif
-  va_end(ap);
-
+  va_end(apSave);
```

Comment:
It is *technically* UB to just assign and reuse va_lists like you did.

If we start from the assumption that a va_list is just a pointer on MSVC and we can just copy it then let's see it through and not use va_end either since it doesn't matter.